transportable-udfs-presto/src/main/java/com/linkedin/transport/presto/StdUdfWrapper.java

```diff
@@ -334,4 +334,24 @@ public Object evalInternal(StdUDF stdUDF, Type[] types, boolean isIntegerReturnT
       Object arg3, Object arg4) {
     return eval(stdUDF, types, isIntegerReturnType, arg1, arg2, arg3, arg4);
   }
+
+  public Object evalInternal(StdUDF stdUDF, Type[] types, boolean isIntegerReturnType, Object arg1, Object arg2,
```

Comment:
Why are these methods public. Is this a limitation of the Java MethodHandle API?

Reply:
Because they replace public methods in the native Presto UDFs. Example: https://github.com/prestodb/presto/blob/master/presto-main/src/main/java/com/facebook/presto/operator/scalar/ArrayConcatFunction.java. Check the method associated with the @UsedByGeneratedCode annotation.

Reply:
Thank you for fixing the 8-param implementation.

ship it

Reply:
Further, they are being called from Presto's: com.facebook.presto.util.Reflection#methodHandle(java.lang.Class<?>, java.lang.String, java.lang.Class<?>...), which expects a public method, and throws NoSuchMethodException on private methods.

Reply:
Thanks Eric!

Reply:
Thanks for checking @wmoustafa